File: backend/src/main/java/com/revnu/backend/features/sales/dto/SaleRequest.java
package com.revnu.backend.features.sales.dto;

import java.math.BigDecimal;
import java.util.UUID;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;

public record SaleRequest(
        @NotNull(message = "Sale amount is required")
        @Positive(message = "Sale amount must be greater than zero")
        BigDecimal amount,
        @NotNull(message = "Category is required")
        UUID categoryId,
        String notes
        ) {

}

File: mobile/app/src/main/java/com/revnu/mobile/features/sales/model/SaleRequest.kt
package com.revnu.mobile.features.sales.model

import com.google.gson.annotations.SerializedName

data class SaleRequest(
    @SerializedName("amount") val amount: Double,
    @SerializedName("categoryId") val categoryId: String,
    @SerializedName("notes") val notes: String?
)